src/charm.py

```diff
@@ -299,17 +371,23 @@ def _on_auth_proxy_config_changed(self, event: AuthProxyConfigChangedEvent) -> N
             )
 
             if rules:
-                # Push the rules to the container
-                filename = f"{self._oathkeeper_access_rules_path}/access-rules-{event.relation_app_name}-{rule_type}.json"
-                self._container.push(filename, str(rules), make_dirs=True)
-                access_rules_locations.append(filename)
+                cm_name = f"access-rules-{event.relation_app_name}-{rule_type}.json"
+                patch = {"data": {cm_name: str(rules)}}
+                self.access_rules_configmap.patch(patch=patch, cm_name="access-rules")
+                access_rules_filenames.append(cm_name)
+
+            self._patch_statefulset(
+                mount_path=self._access_rules_dir_path,
+                name="access-rules",
+                config_map_name=self._access_rules_config_map_name,
+            )
```

Comment:
I don't think this is the right place to patch the statefulset. This will be called multiple times, can't we move it to pebble_ready?